primary/src/grpc_server/configuration.rs

```diff
@@ -76,30 +93,49 @@ impl<PublicKey: VerifyingKey> Configuration for NarwhalConfiguration<PublicKey>
         request: Request<NewNetworkInfoRequest>,
     ) -> Result<Response<Empty>, Status> {
         let new_network_info_request = request.into_inner();
-        let epoch_number = new_network_info_request.epoch_number;
+        // TODO: Figure out how to verify network data against current epoch.
```

Comment:
The epoch is part of the `Committee` struct, so I would say in the simplest form we could do the comparison as part of the endpoint here, e.x:
```
if epoch_number != self.committee.epoch {
  // return error
}
```

otherwise we could refactor the `update_primary_network_info` method to accept the epoch number as well and do the validation there. For example accept the tuple `(Epoch, Stake, PrimaryAddresses)`

Reply:
Yeah, this should work for now. Missed the `epoch` field in the `Committee` struct. Thanks!